pkg/workflows/sdk/runtime.go

```diff
@@ -30,11 +30,13 @@ type Runtime interface {
 }
 
 type FetchRequest struct {
-	URL       string            `json:"url"`                 // URL to query, only http and https protocols are supported.
-	Method    string            `json:"method,omitempty"`    // HTTP verb, defaults to GET.
-	Headers   map[string]string `json:"headers,omitempty"`   // HTTP headers, defaults to empty.
-	Body      []byte            `json:"body,omitempty"`      // HTTP request body
-	TimeoutMs uint32            `json:"timeoutMs,omitempty"` // Timeout in milliseconds
+	URL         string            `json:"url"`                   // URL to query, only http and https protocols are supported.
+	Method      string            `json:"method,omitempty"`      // HTTP verb, defaults to GET.
+	Headers     map[string]string `json:"headers,omitempty"`     // HTTP headers, defaults to empty.
+	Body        []byte            `json:"body,omitempty"`        // HTTP request body
+	TimeoutMs   uint32            `json:"timeoutMs,omitempty"`   // Timeout in milliseconds
+	WithRetries *bool             `json:"withRetries,omitempty"` // Set to true to enable retry logic for requests
+	MaxTries    *uint             `json:"maxTries,omitempty"`    // Max number of times to retry a request, 0 implies no retries
```

Comment:
Can we clarify whether this number includes the first attempt or not -- based on the comment I think not but it's good to be explicit about this.